docs/src/pages/start/contribution-guide.md

```diff
@@ -3,4 +3,276 @@ title: Contribution Guide
+
+## Create new Quasar App Extension
+
+An easy way to start to contribute to Quasar is to generalize the code you have created on your project and publish it as a Quasar App Extension. Follow this guide on how to [create a new extension](../app-extensions/development-guide/introduction).
+
+When done, submit a PR on [Quasar Awesome](https://github.com/quasarframework/quasar-awesome/blob/master/README.md#community-app-extensions), share your achievement via [Quasar Forum](https://forum.quasar-framework.org/category/15/v1-app-extensions). After the review we will [publish your extension](../app-extensions/discover).
+
+## Contributing to the Quasar Core Source Code
+
+As with any project, there are rules to contributing. Ours are written here, please read them carefully. After that, read the [Quasar code of conduct](https://github.com/quasarframework/quasar/blob/dev/.github/CODE_OF_CONDUCT.md) and you’ll be ready to contribute to Quasars’s core repositories.
+
+An article [Look at the source code](https://medium.com/quasar-framework/wip-look-at-the-source-code-please-1b905ea4906) will help you to get familiar with a Quasar code base.
+
+#### Pull Request Guidelines
+
+- The `master` branch is basically just a snapshot of the latest stable release. All development should be done in dedicated branches. **Do not submit PRs against the `master` branch.**
+
+- Checkout a topic branch from the relevant branch, e.g. `dev`, and merge back against that branch.
+
+- **DO NOT** check in `dist` in the commits.
+
+- It's OK to have multiple small commits as you work on the PR - we will let GitHub automatically squash it before merging.
+
+- If adding new feature:
+  - Provide convincing reason to add this feature. Ideally you should open a suggestion issue first and have it greenlighted before working on it.
+
+- If fixing a bug:
+  - If you are resolving a special issue, add `(fix: #xxxx[,#xxx])` (#xxxx is the issue id) in your PR title for a better release log, e.g. `fix: update entities encoding/decoding (fix #3899)`.
+  - Provide detailed description of the bug in the PR. Live demo preferred.
+
+#### Development Setup
+
+You will need [Node.js](http://nodejs.org) **version 8.9+** along [Yarn](https://yarnpkg.com/) or [NPM](https://docs.npmjs.com/getting-started/installing-node). Read `package.json` and take notice of the scripts you can use.
+
+After cloning the repo, in each subfolder run:
+
+``` bash
+$ yarn # or: npm install
+```
+
+#### Commonly used NPM scripts
+
+``` bash
+# Start dev server with a demo app. This app has Quasar source code linked directly so any change will trigger HMR (Hot Module Reload) on the dev server.
+# There's a section for each feature where tests are made.
+$ yarn dev [theme] # or: npm run dev [theme]
```

Comment:
yarn dev [theme] is still a thing?
@rstoenescu

Reply:
Only 2 quick comments ...

1. Does anyone regularly monitor StackOverflow? Even if someone does, should we be directing users there? 

2. The Forum is mentioned before the Discord chat group, yet it appears to me that the main developers hang out mainly in the Discord Chat. Should we place the Discord Chat above the Forum in this document.

Both things are sort of related to ... Wouldn't it be better concentrate our efforts in one single location?